vignettes/rhino-fluent.Rmd

```diff
@@ -0,0 +1,556 @@
+
+So, this is how the UI should look now -
+
+```{r, echo=FALSE}
+knitr::include_graphics("images/rhino_fluent_gif_2.gif")
+```
+
+## 5. Render a Bar Chart with logic in a helper function
+
+We will now add a bar chart that plots the deal amounts closed by the sales reps. For this, we will code a helper function and keep it inside the `app/logic` folder.
+
+### Helper function in the logic folder of the `rhino` project
+
+-   Go to the `app/logic` folder and create a new `.R` file.
+    Lets name it `barchart.R`
+
+-   Install `gglplot2` and `plotly` packages for this purpose.
+    Please follow the same steps that were followed previously during and after new package installation.
+
+-   Here is the code -
+
+    ```         
+    box::use(
+      ggplot2[ggplot, aes, geom_bar, labs, theme_light],
+    )
+
+    #' @export
+    generate_barchart <- function(filtered_data) {
+            ggplot(filtered_data, aes(x = rep_name, y = deal_amount, fill = rep_name)) +
+            geom_bar(stat = "identity") +
+            labs(x = "Sales Rep Name", y = "Deal Amount") +
+            theme_light()
+    }
+    ```
+
+-   The above code block loads the **`ggplot2`** package and imports several functions from it.
+    It then defines a function called **`generate_barchart()`** that takes a **`filtered_data`** argument as input.
+    The function uses the **`ggplot()`** function to create a bar chart with **`x = rep_name`** and **`y = deal_amount`**, where **`rep_name`** and **`deal_amount`** are columns in the **`filtered_data`** data frame.
+    The **`fill`** aesthetic is also set to **`rep_name`**, which maps each bar to a unique color based on the **`rep_name`** variable.
+    The function also adds labels to the x and y axes and applies the **`theme_light()`** theme to the plot.
+    Finally, the function is exported with **`#' @export`**, which makes it available to other packages and scripts that import this one.
+
+### Call the helper function in the desired `rhino` module
+
+-   Now let us go back to the `app\view\datatable.R` file.
+    Here we need to call the `generate_barchart()` function we just created along with the required dependencies to render it on UI as a plotly bar chart.
+
+-   So, the code begins with
+
+    ```         
+    box::use(
+      shiny[moduleServer, NS, uiOutput, renderUI, reactive, div],
+      shiny.fluent[fluentPage, Text, DetailsList, Toggle.shinyInput],
+      tibble[tibble],
+      dplyr[filter],
+      plotly[plotlyOutput, renderPlotly],
+    )
+
+    box::use(
+      app/logic/barchart[generate_barchart],
+    )
+    ```
+
+    Here, we have called the `plotly` library, and also the function that we defined in the `app/logic` folder.
+
+-   Now, we will add one more `div` in the ui section after the datatable to make room for the bar chart
+
+````         
+```         
+ div(
+      class = "ms-depth-64",
+      style = "height: 500px; overflow: auto;",
```

Comment:
Again, avoid inline styling.

Reply:
added comment to redirect user to custom styling in rhino documentation